### ISSUE TITLE
Importing https://gitlab.com/loveshack/sge.git

### ISSUE THREAD

**Issue by danpovey** [son-of-gridengine/sge#1]
Per list discussions here
 http://gridengine.org/pipermail/users/2018-May/thread.html
(search for 'Son of GridEngine succession), I am creating this organization and repo, and attempting to import https://gitlab.com/loveshack/sge.git.  Unfortunately, due to GitLab bugs, that repo has errors ('git fsck' fails due to an issue mentioned here https://stackoverflow.com/questions/21971941/invalid-author-committer-line-missing-space-before-email).
I am running the fix suggested on that page.  It may end up changing the commit hashes.

**Comment by Kunzol** [comment]
Why not just fork it on Gitlab?

**Comment by danpovey** [comment]
Recording of the steps to fix the 'git fsck' failures.  This is after cloning the repo at https://gitlab.com/loveshack/sge.git.

First copy the tags to branches.
```
for t in $(git tag); do git branch  ${t}_branch $t; done
```
Then use the magic invocation to clean up bad commits in all the branches:
```
for b in $(git branch); do git checkout $b; git filter-branch -f --commit-filter 'case "$GIT_AUTHOR_NAME" in *author\<\ email*) GIT_AUTHOR_NAME=author_name; GIT_AUTHOR_EMAIL=author_email;; esac; git commit-tree "$@" '; done
```
It turned out the only tag that ended up getting rewritten was the most recent one, '811'.  Recreating that tag.
```
git tag -f -m "Originally tagged on Jul 13 2013 by Dave Love <d.love@liverpool.ac.uk>.  Re-tagging due to git fsck failure."  811  811_branch
```
Remove all the temporary branches I created:
```
git checkout master; for b in $(git branch |  grep '_branch'); do git branch -D $b; done
```
Prune now-unused objects.  I had to temporarily remove my remotes to get this to work.
```
git remote rm upstream
git remote rm origin
git reflog expire --expire=now --all
git gc  --prune=all
```
now verify that it works:
```
git fsck
```
(no errors).
add back my upstream remote.
```
git remote add upstream git@github.com:son-of-gridengine/sge.git
git push upstream +HEAD:master
for t in $(git tag); do git push upstream $t; done
```

**Comment by 0xaf1f** [comment]
I guess many of the tags got broken in the process. I don't see any of the 8.1.* releases.

**Comment by danpovey** [comment]
There is just '811'.  But that's the same as it was in the source repo, see here:
https://gitlab.com/loveshack/sge/tags

**Comment by danpovey** [comment]
OK, changed them.
documenting it:
```
for n in $(seq 9); do git tag 8.1.$n 81$n; git tag -d 81$n; done
for n in $(seq 9); do git push upstream 8.1.$n; done
for n in $(seq 9); do git push --delete upstream 81$n; done
```

**Comment by addylaszlokovacs** [comment]
OK!

Maybe you also realize..., but on the release page a very old version 
(8.1.1) is showed as an updated version (23 hours ago)...
Wouldn't be better to create a new one (8.1.9.1/8.1.10) from the last 
version (8.1.9) and all the old release let to be untouched?

**Comment by danpovey** [comment]
I had to rewrite the git history (it's documented on 'issue 1') because
github refused to let me upload his repo as-is; it had `git fsck` errors
that could only be fixed by rewriting things.  The content should be the
same as before though.

**Comment by Kunzol** [comment]
I just converted the darcs to git and imported it in my own gitlab. 
I assume the darcs repository is the "master" till now. 

With the help of  [darcs-to-git](https://github.com/purcell/darcs-to-git), where I converted all unknown email addresses of former developer to "unknown@sge.invalid".  

Besides this the size is only 85M (this one on github is 616M). 

    $ du -sh sge sgegithub
    85M     sge
    616M    sgegithub


A diff says it is equal with this one. 

    $ diff -ur -x .git sge sgegithub
    Only in sge: README
    Only in sgegithub: README.md

And the check runs through: 

    $ git fsck
    Checking object directories: 100% (256/256), done.
    Checking objects: 100% (64836/64836), done.

If you like, I can send you the link to the repo via email.
